### PR TITLE
Bugfix/add more unit test for date detection

### PR DESCRIPTION
**Pull request by BastianPoe**
Increases unit test coverage to cover all branches of date detection

### CODE REVIEW

File: src/paperless_tesseract/tests/test_date.py [hunk 1]
```diff
@@ -115,6 +115,42 @@ def test_date_format_7(self):
...
         )
 
+    @mock.patch(
+        "paperless_tesseract.parsers.RasterisedDocumentParser.SCRATCH",
+        SAMPLE_FILES
+    )
+    def test_date_format_8(self):
+        input_file = os.path.join(self.SAMPLE_FILES, "")
+        document = RasterisedDocumentParser(input_file)
+        document.TEXT_CACHE = ("lorem ipsum\n"
```
Comment (danielquinn) [inline src/paperless_tesseract/tests/test_date.py]: The tests are failing 'cause I changed `.TEXT_CACHE` to `._text` in `RasterisedDocumentParser` in keeping with [pep8's rule about constants](https://pep8.org/#constants).

**Comment by danielquinn** [comment]
Thanks for the additional test coverage!  This project can always use more tests :-)  Unfortunately, I did a bunch of hacking on the project this weekend and switched up the RasterisedDocumentParser to be more inline with the style of the rest of the project -- that included the changing of the `.TEXT_CACHE` property you had and a few bits of indentation etc.  If you update from `master` though it shouldn't cause any conflicts or anything.

**Comment by BastianPoe** [comment]
@danielquinn, no problem, just did the changes.

BTW: how do you feel about adding a test coverage analysis such as https://coveralls.io/github/home-assistant/home-assistant to the project? It might help to motivate people to write unit tests.

File: src/paperless_tesseract/tests/test_date.py (hunk 1 above)
Comment (BastianPoe) [reply src/paperless_tesseract/tests/test_date.py]: done

**Comment by danielquinn** [comment]
Honestly, I'm a little daunted by the idea of adding coveralls, as I'm sure the coverage for this project is rather low, but if you think it'll encourage submission of tests, I'll give it a shot :-)

The process of inclusion appears to be a little involved though, so I'll have to save this for when I have some time to figure it out.

Regardless, thanks again for the additional tests!

**Pull request merged**